src/matter/cluster/LevelControlCluster.ts

```diff
@@ -0,0 +1,179 @@
+})
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.4 */
+const features = {
+    /** Dependency with the On/Off cluster */
+    OnOff: BitFlag(0), // TODO default: true.
+
+    /** Behavior that supports lighting applications */
+    Lighting: BitFlag(1),
+
+    /** Supports frequency attributes and behavior. */
+    Frequency: BitFlag(2)
+};
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.2.1 */
+const enum MoveMode {
+    up = 0x0,
+    down = 0x1,
+};
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.3 */
+const enum StepMode {
+    up = 0x0,
+    down = 0x1,
+};
+
+ /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.5 */
+const attributes = {
+    /** the current level of this device */
+    currentLevel: Attribute(0x0,  TlvNullable(TlvUInt8), {default: null}), 
+
+    /** time until the current command is complete in 1/10ths of sec */
+    remainingTime: OptionalAttribute(0x01, TlvUInt16, {default: 0}),
+
+    /** the minimum value of CurrentLevel  */
+    minLevel: OptionalAttribute(0x02, TlvUInt8.bound({ min: 0})),
+
+    /** the maximum value of CurrentLevel */
+    maxLevel: Attribute(0x03, TlvUInt8.bound({ max: 254}), {default: 254}),
+
+    /** the frequency at which the device is at CurrentLevel */
+    currentFrequency: OptionalAttribute(0x04, TlvUInt16, {default: 0}),
+
+    /** min value of CurrentFrequency capable of being assigned */
+    minFrequency: OptionalAttribute(0x05, TlvUInt16.bound({min: 0}), {default: 0}),
+
+    /** max value of CurrentFrequency capable of being assigned */
+    maxFrequency: OptionalAttribute(0x06, TlvUInt16, {default: 0}),
+
+    /** time to move to/from the target when On/Off received by On/Off cluster on the same endpoint */
+    onOffTransitionTime: OptionalAttribute(0x10, TlvUInt16, {default: 0}),
+
+    /** CurrentLevel when OnOff attr of On/Off cluster on the same endpoint is TRUE */
+    onLevel: WritableAttribute(0x11,  TlvNullable(TlvUInt8), {default: null}),
+
+    /** time taken to move the current level from the min level to the max level */
+    onTransitionTime: OptionalWritableAttribute(0x12, TlvNullable(TlvUInt16,), {default: null}),
+
+    /** time taken to move the current level from the max level to the minlevel  */
+    offTransitionTime: OptionalWritableAttribute(0x13, TlvNullable(TlvUInt16), {default: null}),
+
+    /** default rate in units per second when a Move command has a null Rate */
+    defaultMoveRate: OptionalWritableAttribute(0x14, TlvNullable(TlvUInt8)),
+
+    options: Attribute(0x0f,  TlvBitmap(TlvUInt8, OptionsBitmap) ),
+
+    startUpCurrentLevel: OptionalAttribute(0x4000, TlvNullable(TlvUInt8),),  
+};
+
+/* TODO - Implement all below and change from Optional to Mandatory */
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6 */
+const commands = {
+    /** Move from the current level to the specified level */
```

Comment:
nit: Method jsdocs are usually an action starting with a verb at the 3rd person.
Moves from the current level to the specified level.
(starting with a capital letter and ending with a dot)

Same for comments below.